src/main/webapp/app/exercises/text/assess/text-assessment-base.component.ts

```diff
@@ -44,7 +44,7 @@ export abstract class TextAssessmentBaseComponent implements OnInit {
 
     protected computeTotalScore(assessments: Feedback[]): number {
         // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
-        const maxPoints = this.exercise?.maxPoints! + this.exercise?.bonusPoints! ?? 0.0;
+        const maxPoints = this.exercise?.maxPoints! + (this.exercise?.bonusPoints ?? 0.0);
```

Comment:
Here you assume `bonusPoints` can be undefined. According to the changes in e.g. `file-upload-assessment.component.ts` it isn't. If bonusPoints can be undefined, we shouldn't handle it as defined above.

Reply:
Moved to exercise model